packages/contract-wrappers/src/contract_wrappers/coordinator_wrapper.ts

```diff
@@ -0,0 +1,857 @@
+        for (const order of signedOrders) {
+            const feeRecipient = order.feeRecipientAddress;
+            if (feeRecipientsToOrders[feeRecipient] === undefined) {
+                feeRecipientsToOrders[feeRecipient] = [] as SignedOrder[];
+            }
+            feeRecipientsToOrders[feeRecipient] = feeRecipientsToOrders[feeRecipient].concat(order);
+        }
+
+        const serverEndpointsToFeeRecipients: { [feeRecipient: string]: string[] } = {};
+        for (const feeRecipient of Object.keys(feeRecipientsToOrders)) {
+            const endpoint = await this._getServerEndpointOrThrowAsync(feeRecipient);
+            if (serverEndpointsToFeeRecipients[endpoint] === undefined) {
+                serverEndpointsToFeeRecipients[endpoint] = [];
+            }
+            serverEndpointsToFeeRecipients[endpoint] = serverEndpointsToFeeRecipients[endpoint].concat(feeRecipient);
+        }
+
+        // make server requests
+        let numErrors = 0;
+        let errorResponses: CoordinatorServerResponse[] = [];
+        let approvalResponses: CoordinatorServerResponse[] = [];
+        const transaction = await this._generateSignedZeroExTransactionAsync(data, takerAddress);
+        for (const endpoint of Object.keys(serverEndpointsToFeeRecipients)) {
+            const response = await this._executeServerRequestAsync(transaction, takerAddress, endpoint);
+            if (response.isError) {
+                errorResponses = errorResponses.concat(response);
+                numErrors++;
+            } else {
+                approvalResponses = approvalResponses.concat(response);
+            }
+        }
+
+        // if no errors
+        if (numErrors === 0) {
+            // concatenate all approval responses
+            const allApprovals = approvalResponses
+                .map(resp => formatRawResponse(resp.body as CoordinatorServerApprovalRawResponse))
+                .reduce((previous, current) => {
+                    previous.signatures = previous.signatures.concat(current.signatures);
+                    previous.expirationTimeSeconds = previous.expirationTimeSeconds.concat(
+                        current.expirationTimeSeconds,
+                    );
+                    return previous;
+                });
+
+            // submit transaction with approvals
+            return this._submitCoordinatorTransactionAsync(
+                transaction,
+                takerAddress,
+                transaction.signature,
+                allApprovals.expirationTimeSeconds,
+                allApprovals.signatures,
+                orderTransactionOpts,
+            );
+        } else {
+            // format errors and approvals
+            // concatenate approvals
+            const approvedOrders = approvalResponses
+                .map(resp => {
+                    const endpoint = resp.coordinatorOperator;
+                    const feeRecipients: string[] = serverEndpointsToFeeRecipients[endpoint];
```

Comment:
Is this type declaration `string[]` necessary?

Reply:
nope, removing